test/libsolutil/StringUtils.cpp

```diff
@@ -210,6 +210,33 @@ BOOST_AUTO_TEST_CASE(test_format_number_readable_signed)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(test_encodeURI)
```

Comment:
It would be nice to add tests for all of the 'special' characters, i.e. non alpha-numeric from `allowedCharacters` list. Also, can you add some realistic examples here in the context of the LSP (both for encoding and decoding)?

Reply:
Ah, I see @Marenz already left a similar top level comment.